frontend/ui/lib/services/rest_api_place_visit_service.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:convert';
+import 'package:tripmeout/model/place_visit.dart';
+import 'package:http/http.dart' as http;
+import 'package:tripmeout/services/place_visit_service.dart';
+
+class RestApiPlaceVisitService implements PlaceVisitService {
+  final String endpoint;
+
+  RestApiPlaceVisitService({this.endpoint = ''});
+
+  Future<List<PlaceVisit>> listPlaceVisits(String tripid) async {
+    final response = await http.get('$endpoint/api/trips/$tripid/placeVisits');
+    if (response.statusCode == 200) {
+      var asList =
+          (json.decode(response.body) as List).cast<Map<String, dynamic>>();
+      return asList.map((x) => PlaceVisit.fromJson(x)).toList();
+    } else {
+      throw Exception(
+          'Failed to get List of PlaceVisits with status code ${response.statusCode}');
+    }
+  }
+
+  Future<PlaceVisit> getPlaceVisit(String tripid, String id) async {
+    final response =
+        await http.get('$endpoint/api/trips/$tripid/placeVisits/$id');
+    if (response.statusCode == 200) {
+      return PlaceVisit.fromJson(json.decode(response.body));
+    } else {
+      throw Exception(
+          'Failed to get PlaceVisit with status code ${response.statusCode}');
+    }
+  }
+
+  Future<PlaceVisit> createPlaceVisit(PlaceVisit placeVisit) async {
+    String tripid = placeVisit.tripid;
+    final http.Response response = await http.post(
+      '$endpoint/api/trips/$tripid/placeVisits',
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+      },
+      body: jsonEncode(<String, dynamic>{
```

Comment:
Do we need to manually construct the serializable Map here? We should move this logic so it lives alongside our deserialization code. Doing this inline (in two separate places) means that we have to keep both of these implementations in sync with the deserializer.